### ISSUE TITLE
Changing Image in Preview window enhancements

### ISSUE THREAD

**Issue by MasterPetrik** [Bionus/imgbrd-grabber#882]
There is some suggestions to solve Changing Image in Preview window problems.

# The case 1: 
**In preview window taglist and window caption changing much earlier than image itself is loaded.** 

This situation is cause a misunderstanding while fast scrolling images in preview window and even if you just going to the next image, and it's not loaded yet, you will see it's taglist and window caption, but old image in the middle of the window. That's it. Old image still here. Your mind wispers you "Maybe I misclicked? Why image doesn't changed? Looks like I need to press "next" button again."

So I suggest you a simple solution: When going to the next image, change all info in preview window(image taglist, window caption, details widow info) ONLY after at least starting of loading image preview. Not erlier. 

# The case 2: 

I noticed, if loading of image in preview window has been going on for a long time, there are nothing showed, even if image will not be loaded at all. This is not optimal solution. I suggest you, if loading of image in preview window has been going on for a long time(more then 2-3 sec) then show it's thumbnail(of course rescaled to image sizes) from search results grid. It's important to load thumblail image for that from already downloaded thumblail from grabber's temp folder, instead of loading it from site too.

**Comment by Bionus** [comment]
> So I suggest you a simple solution: When going to the next image, change all info in preview window(image taglist, window caption, details widow info) ONLY after at least starting of loading image preview. Not erlier.

Well if you do this, that means nothing happens when you click "next" (until the image is loaded). So the problem is even more visible and the user will just press "next" again, loading the n+2 image.

A simplier solution is to remove the image when clicking next.

> I suggest you, if loading of image in preview window has been going on for a long time(more then 2-3 sec) then show it's thumbnail

Isn't there already the thumbnail shown when you click next?

**Comment by MasterPetrik** [comment]
> Well if you do this, that means nothing happens when you click "next" (until the image is loaded). So the problem is even more visible and the user will just press "next" again, loading the n+2 image.

True.

> A simplier solution is to remove the image when clicking next.

remove the image and look at blank space for some seconds? Hehe... no.
A REALLY simplier solution is to showing next image upscaled thumbnail when clicking next. With changing caption and taglist. So it will be obvious that new image is coming.

> I suggest you, if loading of image in preview window has been going on for a long time(more then 2-3 sec) then show it's thumbnail

> Isn't there already the thumbnail shown when you click next?

No. That works for some images, and for some images it isn't works as I described, that's why I wrote all that text in Case 2. For example, if you faced with problem image from https://github.com/Bionus/imgbrd-grabber/issues/881 issue, you will not get thumbnail instead of not loaded image, you will see it's taglist and window caption, but previous image in the middle of the window, so you will get no image, no thumbnail - nothing exept tags and window caption. -___-
